backends/platform/libretro/scripts/bundle_datafiles.sh

```diff
@@ -0,0 +1,163 @@
+read -a DATAFILES_ARRAY -d '' -r <<< "$DATAFILES_LIST"
+set -e
+
+# Add specific data files
+DATAFILES_ARRAY[${#DATAFILES_ARRAY[@]}]="${SCUMMVM_PATH}"/backends/vkeybd/packs/vkeybd_default.zip
+
+# Make sure target folders exist
+[ $3 = "bundle" ] && mkdir -p "${TMP_PATH}/${BUNDLE_THEME_DIR}/"
+[ $3 = "bundle" ] && mkdir -p "${TMP_PATH}/${BUNDLE_DATAFILES_DIR}/"
+
+count=0
+# Process themes
+	process_group "$BUNDLE_THEME_DIR" $3 ${THEME_ARRAY[@]}
+
+# Process datafiles
+	process_group "$BUNDLE_DATAFILES_DIR" $3 ${DATAFILES_ARRAY[@]}
+
+# Process additional local bundle files
+if [ -d "$BUNDLE_LOCAL_DATAFILES_DIR" -a ! -z "$(ls -A ${BUNDLE_LOCAL_DATAFILES_DIR} 2>/dev/null)" ] ; then
+	for item in $BUNDLE_LOCAL_DATAFILES_DIR/*; do
+		[ ! $(echo "$item" | sed "s|^.*/||g") = "README.md" ] && LOCAL_EXTRA_ARRAY+=("$item")
+	done
+	process_group "$BUNDLE_DATAFILES_DIR" $3 ${LOCAL_EXTRA_ARRAY[@]}
+fi
+
+if [ ! $3 = "bundle" ]; then
+
+# Update from libretro ScummVM.dat
+wget -NO "$BUILD_PATH"/ScummVM.dat https://raw.githubusercontent.com/libretro/libretro-database/master/dat/ScummVM.dat
+[ -f "$BUILD_PATH"/ScummVM.dat ] && SUPPORTED_EXTENSIONS="$(cat $BUILD_PATH/ScummVM.dat | grep 'rom (' | sed -e 's/\" .*//g' -e 's/.*\.//g' | sort -u | tr '\n' '|')" || SUPPORTED_EXTENSIONS="$ALLOWED_EXT"
+
+	# Create core.info file
+	set +e
+	read -d '' CORE_INFO_CONTENT <<EOF
+# Software Information
+display_name = "$NICE_NAME"
+authors = "SCUMMVMdev"
+supported_extensions = "$SUPPORTED_EXTENSIONS"
+corename = "$NICE_NAME"
+categories = "Game"
+license = "GPLv3"
+permissions = ""
+display_version = $(cat $SCUMMVM_PATH/base/internal_version.h 2>/dev/null | grep SCUMMVM_VERSION | sed "s|^.*SCUMMVM_VERSION *||g")
+
+# Hardware Information
+manufacturer = "Various"
+systemname = "Game engine"
+systemid = "scummvm"
+
+# Libretro Features
+database = "ScummVM"
+supports_no_game = "true"
+savestate = "false"
+savestate_features = "null"
+cheats = "false"
+input_descriptors = "true"
+memory_descriptors = "false"
+libretro_saves = "false"
+core_options = "true"
+core_options_version = "1.3"
+load_subsystem = "false"
+hw_render = "false"
+needs_fullpath = "true"
+disk_control = "false"
+is_experimental = "false"
+
+# Firmware / BIOS
+firmware_count = $count
+EOF
+	set -e
+
+	CORE_INFO_CONTENT="${CORE_INFO_CONTENT}${CORE_INFO_DATS}
+description = \"The ScummVM adventure game engine ported to libretro. This core is built directly from the upstream repo and is synced upon stable releases, though it is not supported upstream. So please report any bug to Libretro and/or make sure the same apply to the standalone ScummVM program as well, before making any report to ScummVM Team.\""
+	echo "$CORE_INFO_CONTENT" > "${TARGET_PATH}/${INFO_FILE_PRE}_libretro.info"
+	echo "${INFO_FILE_PRE}_libretro.info created successfully"
+else
+
+	# Create archive
+	rm -f "${TARGET_PATH}/$BUNDLE_ZIP_FILE"
+	cd "${TMP_PATH}"
+	zip -rq "${TARGET_PATH}/$BUNDLE_ZIP_FILE" "${BUNDLE_DIR}" > /dev/null 2>&1
```

Comment:
I'd say that this could be a noticeable performance hit, because zip will have to extract the files into memory and some files like `fonts.dat` are really big.

I would suggest using some different method with simple file concatenation. Moreover, since most of our data files are compressed anyway, double compression will not bring you size gains.